runtime/win32.c

```diff
@@ -1154,6 +1156,151 @@
     CAMLassert(0);
 }
 
+/* Threads */
+
+int caml_plat_thread_create(caml_plat_thread *restrict thread,
+                            const caml_plat_thread_attr *restrict attr,
+                            unsigned ( WINAPI *start_address )( void * ),
+                            void *restrict arg)
+{
+  (void) attr; /* unused */
+  *thread = (caml_plat_thread)(uintptr_t) _beginthreadex(
```

Comment:
At least for my education, what is the purpose of the double cast in `(caml_plat_thread)(uintptr_t) _beginthreadex`, especially as `_beginthreadex` returns a `uintptr_t` in the first place, if I read the documentation correctly?

Reply:
It's a problem with clang-cl complaining about converting pointers and integers, even though `uintptr_t` is wide enough to hold a pointer. I'll add the error message to the commit message or the PR to explain this.

Reply:
I couldn't reproduce the error, I must have used a stronger set of warnings than the default. I'm removing this double-cast for now.